File: Data Structures & Algorithms/longest-consecutive-sequence/submission-18.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const res = new Set(nums)
        let longest = 0

        for (let i = 0; i < nums.length; i++) {
            if (!res.has(nums[i] - 1)) {
                let length = 1
                while (res.has(nums[i] + 1)) {
                    length++
                    nums[i]++
                }
                longest = Math.max(longest, length)
            }
        }
        return longest
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-18.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        longest = 0
        maxf = 0
        l, r = 0, 0
        count = {}

        while r < len(s):
            count[s[r]] = 1 + count.get(s[r], 0)
            maxf = max(maxf, count.get(s[r]))

            while (r - l + 1) - maxf > k:
                count[s[l]] -= 1
                l += 1
            longest = max(longest, r - l + 1)
            r += 1
        return longest